### DISCUSSION TITLE
Configuration for the possibility of using postgresql

### DISCUSSION THREAD

**Discussion by WalterGit** [epicweb-dev/epic-stack#700] [Ideas]
It would be interesting to add an environment variable or dynamically retrieve the provider that is being used in Prisma, because when trying to use Prism in the stack as it is, it generates several errors.

**Comment by kentcdodds** [comment]
Hi @WalterGit!

I think this would make a great [example](https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md), but it wouldn't work well as a configuration option. It would complicate the codebase too much and most people will only want one or the other, not the ability to easy switch between the two.

You can definitely start with the stack and migrate it to postgres if you like, but that's not something I have time to build myself as an example. Sorry!

**Comment by andreaspapav** [comment]
@kentcdodds just created a project using Epic Stack and connected it to a postgresql db. Basically just updated the schema.prisma and created a docker-compose.yml to spin up a dev and test db. Although everything works great with the app i cant get the tests setup correct. It still seems to be trying to connect  to the sqlite db. If we get the vitest and playwright tests to run i will add it to the examples page.

**Reply by kentcdodds** [reply]
Yeah, you're gonna need to change the way testing is done. You will likely not want your tests to interact directly with the database or you'll need to change the setup a bit so instead of each test getting its own database you somehow share and cleanup a test database.

The testing setup will be quite different if you're not using SQLite.
